includes/class-wc-regenerate-images-request.php

```diff
@@ -146,15 +146,15 @@ protected function task( $item ) {
 	 * @param int   $attachment_id Attachment ID.
 	 * @return array An associative array of image sizes.
 	 */
-	public function filter_image_sizes_to_only_missing_thumbnails( $sizes, $metadata, $attachment_id ) {
+	public function filter_image_sizes_to_only_missing_thumbnails( $sizes, $metadata, $attachment_id = null ) {
 		if ( ! $sizes ) {
 			return $sizes;
 		}
 
 		$fullsizepath = get_attached_file( $attachment_id );
 		$editor       = wp_get_image_editor( $fullsizepath );
 
-		if ( is_wp_error( $editor ) ) {
+		if ( is_wp_error( $editor ) || is_null( $attachment_id ) ) {
```

Comment:
The is_null check needs to happen almost right at the top as $attachment_id is used above this check which might cause errors/notices.

Reply:
Ah good catch, totally missed that it's used before.